click/_compat.py

```diff
@@ -33,6 +34,12 @@ def _make_text_stream(stream, encoding, errors,
                                     force_readable=force_readable,
                                     force_writable=force_writable)
 
+def _get_current_umask():
+    """Get current umask."""
+    umask = os.umask(0)
```

Comment:
This is, unfortunately, a potential security problem in the presence of multiple threads. (See https://bugs.python.org/issue21082.)

I’m not sure what to suggest instead, other than avoiding `tempfile` and writing the loop to safely create a fresh filename manually.